ci-operator/config/openshift/installer/openshift-installer-master.yaml

```
@@ -250,17 +253,19 @@ tests:
     cluster_profile: azure4
     workflow: openshift-e2e-azure-resourcegroup
 - as: e2e-aws-upgrade
-  skip_if_only_changed: (^docs/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Z]+\.md$)|(^(data/.*|pkg/.*|images)/(azure|baremetal|gcp|libvirt|openstack|ovirt|vsphere)/)
+  skip_if_only_changed: (^(docs|upi|hack)/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Z]+\.md$)|(^(data/.*|pkg/.*|images)/(alibabacloud|azure|baremetal|gcp|ibmcloud|libvirt|nutanix|openstack|ovirt|powervs|vsphere)/)
   steps:
     cluster_profile: aws
     workflow: openshift-upgrade-aws-latestinstaller
 - as: e2e-aws-upi
   optional: true
+  skip_if_only_changed: (^(docs|data|pkg|images|hack)/)|((^|/)OWNERS(_ALIASES)?$)|((^|/)[A-Z]+\.md$)|(^(upi/.*)/(alibabacloud|azure|azurestack|baremetal|gcp|ibmcloud|libvirt|nutanix|openstack|ovirt|powervs|vsphere)/)
```

What about: 
```suggestion
  run_if_changed: ^(upi/aws/.*)$
```
and follow that pattern for all upi jobs. please double check the regex as I'm not super confident with that